utils/zero_wipe.go

```diff
@@ -53,45 +52,56 @@ func (z *ZeroWipe) Wipe(ctx context.Context, logicalName string) error {
 	// Create a buffer fill with zeroes
 	buffer := make([]byte, bufferSize)
 
-	var bytesWritten int64
+	var totalBytesWritten int64
+	var bytesSinceLastPrint int64
 	start := time.Now()
 
-	// rewind cassete tape ;=)
-	file.Seek(0, 0)
+	// rewind cassette tape ;=)
+	whence = 0 // 0 means relative to the origin of the file, check Seek function
+	_, err = file.Seek(0, whence)
+	if err != nil {
+		log.Println(err)
+		return err
+	}
 
-	// Writting zeroes loop
-	i := 0
-	for bytesWritten < partitionSize {
-		n, err := file.Write(buffer)
+	// Writing zeroes loop
+	bytesSinceLastPrint = 0
+	for totalBytesWritten < partitionSize {
+		bytesWritten, err := file.Write(buffer)
 		if err != nil {
-			if strings.Contains(err.Error(), "no space left on device") { //syscall.ENOSPC
-				// If partitionSize % bufferSize is not 0 the last write loop will not write all the buffer size
-				log.Println("we have reached the end of the device.")
+			if strings.Contains(err.Error(), "no space left on device") { // syscall.ENOSPC
+				// Check if the last write loop didn't write all the buffer size, the reason is partitionSize % bufferSize is not 0
+				if totalBytesWritten+int64(bytesWritten) == partitionSize {
+					log.Println("we have reached the end of the devices.")
+					return nil
+				} else {
+					log.Println("we have a problem with the write to disk: ", err)
+				}
 			} else {
 				// Other errors
-				log.Println("failed to write to disk:", err)
+				log.Println("failed to write to disk: ", err)
 			}
 			return err
 		}
-		bytesWritten += int64(n)
-		i++
-		// Calculate progress and ETA
-		progress := float64(bytesWritten) / float64(partitionSize) * 100
-		elapsed := time.Since(start).Seconds()
-		speed := float64(n*i) / elapsed                                              // Speed in bytes per second
-		remainingSeconds := (float64(partitionSize) - float64(bytesWritten)) / speed // Remaining time in seconds
-		remainingHours := float64(remainingSeconds / 3600)
-		mbPerSecond := speed / (1024 * 1024)
+		totalBytesWritten += int64(bytesWritten)
+		bytesSinceLastPrint += int64(bytesWritten)
 
 		// Print progress each 10 seconds
 		if time.Since(start) >= 10*time.Second {
+			// Calculate progress and ETA
+			progress := float64(totalBytesWritten) / float64(partitionSize) * TotalPercentage
```

Comment:
The progress printer would be easier to understand if its moved into its own function

Reply:
I like it!